File: app/src/main/java/com/vsa/filmoteca/data/usecase/MoviesPersistanceUseCase.java
package com.vsa.filmoteca.data.usecase;

import com.vsa.filmoteca.data.domain.Movie;
import com.vsa.filmoteca.data.repository.MoviesPersistanceRepository;

import java.util.List;

/**
 * Created by albertovecinasanchez on 22/7/16.
 */

public class MoviesPersistanceUseCase {

    private MoviesPersistanceRepository mMoviesPersistanceRepository;

    public MoviesPersistanceUseCase(MoviesPersistanceRepository moviesPersistanceRepository) {
        mMoviesPersistanceRepository = moviesPersistanceRepository;
    }

    public int getCurrentMovieIndex() {
        return mMoviesPersistanceRepository.getCurrentMovieIndex();
    }

    public void setCurrentMovieIndex(int index) {
        mMoviesPersistanceRepository.setCurrentMovieIndex(index);
    }

    public int getMoviesCount() {
        return mMoviesPersistanceRepository.getMoviesCount();
    }

    public void setMoviesCount(int size) {
        mMoviesPersistanceRepository.setMoviesCount(size);
    }

    public Movie getCurrentMovie() {
        return mMoviesPersistanceRepository.getCurrentMovie();
    }

    public void setMovies(List<Movie> movies) {
        mMoviesPersistanceRepository.setMovies(movies);
    }

}

File: app/src/main/java/com/vsa/filmoteca/data/usecase/MoviesPersistanceUseCase.kt
package com.vsa.filmoteca.data.usecase

import com.vsa.filmoteca.data.domain.Movie
import com.vsa.filmoteca.data.repository.MoviesPersistanceRepository
import javax.inject.Inject

/**
 * Created by albertovecinasanchez on 22/7/16.
 */

class MoviesPersistanceUseCase @Inject constructor(private val repository: MoviesPersistanceRepository) {

    var currentMovieIndex: Int
        get() = repository.currentMovieIndex
        set(index) {
            repository.currentMovieIndex = index
        }

    var moviesCount: Int
        get() = repository.moviesCount
        set(size) {
            repository.moviesCount = size
        }

    val currentMovie: Movie
        get() = repository.currentMovie

    fun setMovies(movies: List<Movie>) {
        repository.setMovies(movies)
    }

}
